app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -1,36 +1,89 @@
+import { Input, Textarea, Select } from "neetoui/formik";
+import { Button, DateInput, Switch, Toastr } from "neetoui";
 
 export default function NewNoteForm({ onClose, refetch }) {
-  const handleSubmit = async values => {
+  const [dueDateCheck, setDueDateCheck] = useState(false);
+  const [selectedDueDate, setSelectedDueDate] = useState(new Date());
+
+  const handleSubmit = async () => {
     try {
-      await notesApi.create(values);
       refetch();
       onClose();
+      Toastr.success("New note added successfully");
     } catch (err) {
       logger.error(err);
     }
   };
+
+  const tagOptions = [
+    { label: "Internal", value: "internal" },
+    { label: "External", value: "external" },
+  ];
+
+  const contactOptions = [
+    { label: "Amal Dinesh", value: "amaldinesh" },
+    { label: "Karthik Menon", value: "karthikmenon" },
+  ];
   return (
     <Formik
       initialValues={{
-        title: "",
-        description: "",
+        title: "React Onboarding Challenge",
+        tags: tagOptions[0],
+        description: "Please use neeto-ui and tailwindCSS.",
+        assignedContact: contactOptions[0],
       }}
       onSubmit={handleSubmit}
       validationSchema={yup.object({
         title: yup.string().required("Title is required"),
+        tags: yup.object().required("Tag is required"),
         description: yup.string().required("Description is required"),
+        assignedContact: yup.object().required("Contact is required"),
       })}
     >
       {({ isSubmitting }) => (
         <Form>
-          <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Input label="Note Title" name="title" className="mb-6" />
+          <Select
+            label="Tags"
+            isCreateable
+            defaultValue={[{ value: "internal", label: "Internal" }]}
+            placeholder="Select an Option"
+            name="tags"
+            defaultOptions={tagOptions}
+            className="mb-6"
+          />
+          <Textarea
+            label="Note Description"
+            name="description"
+            rows={8}
+            className="mb-6"
+          />
+          <Select
+            label="Assigned Contact"
+            isCreateable
+            defaultValue={{ value: "internal", label: "Internal" }}
+            placeholder="Select a Contact"
+            name="assignedContact"
+            defaultOptions={contactOptions}
+          />
+           
+          <div className="flex justify-between mb-6">
+            <label>Add Due Date to Note</label>
+            <Switch
+              checked={dueDateCheck}
+              onChange={e => setDueDateCheck(e.target.value)}
+            />
+          </div>
+          {dueDateCheck && (
+            <DateInput
+              value={selectedDueDate}
+              label="Due Date"
+              onChange={newDate => setSelectedDueDate(newDate)}
+            />
+          )}
```

Comment:
Instead of using `mb-6` for each block individually, please use `space-x-6` on the parent container.
Please refer [space between](https://tailwindcss.com/docs/space#class-reference) for more details.